x-pack/test/observability_ai_assistant_functional/tests/contextual_insights/index.spec.ts

```diff
@@ -0,0 +1,163 @@
+            .failure()
+        ),
+    ];
+
+    await apmSynthtraceEsClient.index(documents);
+  }
+
+  async function createOpenAiConnector(proxy: LlmProxy) {
+    await supertest
+      .post('/api/actions/connector')
+      .set('kbn-xsrf', 'foo')
+      .send({
+        name: 'foo',
+        config: {
+          apiProvider: 'OpenAI',
+          apiUrl: `http://localhost:${proxy.getPort()}`,
+          defaultModel: 'gpt-4',
+        },
+        secrets: { apiKey: 'myApiKey' },
+        connector_type_id: '.gen-ai',
+      })
+      .expect(200);
+  }
+
+  async function deleteConnectors() {
+    const connectors = await supertest.get('/api/actions/connectors').expect(200);
+    const promises = connectors.body.map((connector: { id: string }) => {
+      return supertest
+        .delete(`/api/actions/connector/${connector.id}`)
+        .set('kbn-xsrf', 'foo')
+        .expect(204);
+    });
+
+    try {
+      await Promise.all(promises);
+    } catch (e) {
+      if (e?.response?.status === 404) {
+        return;
+      }
+
+      log.debug('Error deleting connector');
+      throw e;
+    }
+  }
+
+  async function navigateToError() {
+    await common.navigateToApp('apm', {
+      hash: `/services/opbeans-go/errors?rangeFrom=now-15m&rangeTo=now`,
+    });
+
+    await testSubjects.click('errorGroupId');
+  }
+
+  describe('Contextual insights for APM errors', () => {
+    let proxy: LlmProxy;
+
+    before(async () => {
+      await Promise.all([
+        deleteConnectors(), // cleanup previous connectors
+        apmSynthtraceEsClient.clean(), // cleanup previous synthtrace data
+        ui.auth.login(), // login
+      ]);
+
+      await generateErrorData();
+    });
+
+    describe('when there are no connectors', () => {
+      it('should not show the contextual insight component', async () => {
+        await navigateToError();
+        await retry.try(async () => {
+          await testSubjects.missingOrFail('obsAiAssistantInsightButton');
+        });
+      });
+    });
+
+    describe('when there are connectors', () => {
+      before(async () => {
+        proxy = await createLlmProxy(log);
+        await createOpenAiConnector(proxy);
+      });
+
+      it('should show the contextual insight component on the APM error details page', async () => {
+        await navigateToError();
+
+        proxy
+          .intercept(
+            'conversation',
+            (body) => !isFunctionTitleRequest(body),
+            'This error is nothing to worry about. Have a nice day!'
+          )
+          .complete();
+
+        await retry.try(async () => {
+          await testSubjects.click('obsAiAssistantInsightButton');
+          const llmResponse = await testSubjects.getVisibleText('obsAiAssistantInsightResponse');
+          expect(llmResponse).to.contain('This error is nothing to worry about. Have a nice day!');
+        });
+
+        await proxy.waitForAllInterceptorsSettled();
+      });
+
+      after(async () => {
```

Comment:
you should also delete the connectors here no? and clean the APM index?

Reply:
I should! Oversight on my part during testing